client/findproviders.go

```diff
@@ -0,0 +1,119 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-delegated-routing/parser"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+func (c *client) FindProviders(ctx context.Context, cid cid.Cid) ([]peer.AddrInfo, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	ch, err := c.FindProvidersAsync(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+	select {
+	case r := <-ch:
+		cancel()
+		if r.Err != nil {
+			return nil, err
+		} else {
+			return r.AddrInfo, nil
+		}
```

Comment:
This should be an aggregator instead of just returning after the first one. If we're only going to handle a single response then we don't really need the async function.